### PR TITLE
Initialize arrays of errors / error messages + minor refactoring

### LINKED ISSUE #1681: Crash with 1.7.1 - nil pointer
- **Which image of the operator are you using?**  registry.opensource.zalan.do/acid/postgres-operator:v1.7.1
- **Where do you run it - cloud or metal? Kubernetes or OpenShift?** Bare Metal K8s
- **Are you running Postgres Operator in production?** yes
- **Type of issue?** Bug report

```
time="2021-11-09T12:43:55Z" level=error msg="could not sync roles: could not init db connection: could not init db connection: pq: password authentication failed for user \"postgres\"" cluster-name=harbor/harbor-db pkg=cluster
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x1a4d62e]

goroutine 171 [running]:
github.com/zalando/postgres-operator/pkg/cluster.(*Cluster).syncConnectionPooler(0xc000c1e000, 0xc002c1a800, 0xc002c1b400, 0xc002127630, 0xc002127501, 0x1, 0x1, 0xc000681000, 0x0)
	/workspace/pkg/cluster/connection_pooler.go:715 +0xd6e
github.com/zalando/postgres-operator/pkg/cluster.(*Cluster).Update(0xc000c1e000, 0xc002c1a800, 0xc002c1b400, 0x0, 0x0)
	/workspace/pkg/cluster/cluster.go:853 +0x7da
github.com/zalando/postgres-operator/pkg/controller.(*Controller).processEvent(0xc000966000, 0xc05ab8d65b181b2e, 0x907c75befb, 0x30e4920, 0xc001306990, 0x24, 0x1fdea56, 0x6, 0xc002c1a800, 0xc002c1b400, ...)
	/workspace/pkg/controller/postgresql.go:255 +0x1405
github.com/zalando/postgres-operator/pkg/controller.(*Controller).processClusterEventsQueue(0xc000966000, 0x1, 0xc0001c10e0, 0xc000609490)
	/workspace/pkg/controller/postgresql.go:344 +0x278
created by github.com/zalando/postgres-operator/pkg/controller.(*Controller).Run
	/workspace/pkg/controller/controller.go:440 +0x8a
```

### PR DESCRIPTION
**Pull request by FxKu**
1. In different places we are collecting errors in an array. It should be initialized correctly with make.
2. Minor refactoring in `patroni_test` 
3. fixes #1681

### CODE REVIEW

**Comment by FxKu** [comment]
Might have introduced a bug. e2e tests always fail at:
```
======================================================================
FAIL: test_infrastructure_roles (test_e2e.EndToEndTestCase)
Test using external secrets for infrastructure roles
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/timeout_decorator/timeout_decorator.py", line 81, in new_function
    return function(*args, **kwargs)
  File "/tests/test_e2e.py", line 590, in test_infrastructure_roles
    self.eventuallyTrue(verify_role, "infrastructure role setup is not loaded")
  File "/tests/test_e2e.py", line 65, in eventuallyTrue
    self.assertTrue(f(), m)
AssertionError: False is not true : infrastructure role setup is not loaded
```

File: pkg/controller/util.go [hunk 1]
```diff
@@ -230,9 +229,7 @@ func (c *Controller) getInfrastructureRoles(
 			continue
 		}
 
-		for _, r := range infraRoles {
-			roles = append(roles, r)
-		}
+		roles = append(roles, infraRoles...)
```
Comment (FxKu) [inline pkg/controller/util.go:232]:
```suggestion
		for _, r := range infraRoles {
			roles = append(roles, r)
		}
```

**Comment by RafiaSabih** [comment]
:+1:

**Comment by FxKu** [comment]
:+1:

**Pull request merged**